reactor-core/src/main/java/reactor/core/publisher/FluxLimitRequest.java

```diff
@@ -31,11 +32,18 @@
 
 	FluxLimitRequest(Flux<T> flux, long cap) {
 		super(flux);
+		if (cap < 0) {
+			throw new IllegalArgumentException("cap >= 0 required but it was " + cap);
+		}
 		this.cap = cap;
 	}
 
 	@Override
```

Comment:
nitpick: Can we add `@Nullable` so that we know that `null` is acceptable outcome?

Reply:
fixed